src/components/H3Grid.tsx

```diff
@@ -0,0 +1,85 @@
+import MapboxGL, { LineLayerStyle } from '@react-native-mapbox-gl/maps'
+import React, { memo, useMemo } from 'react'
+import { Feature, FeatureCollection, Position } from 'geojson'
+import geojson2h3 from 'geojson2h3'
+import { StyleProp } from 'react-native'
+import { Colors } from '../theme/theme'
+import { useColors } from '../theme/themeHooks'
+
+type Props = {
+  bounds?: Position[]
+  res?: number
+  color?: Colors
+  width?: number
+  visible?: boolean
+  zoomLevel: number
+}
+
+const H3Grid = ({
+  bounds,
+  res = 8,
+  color = 'blueDarkest',
+  width = 1,
+  visible = true,
+  zoomLevel,
+}: Props) => {
+  const colors = useColors()
+  const styles = useMemo(() => makeStyles(colors[color], width), [
+    color,
+    colors,
+    width,
+  ])
+
+  const boundingBox = useMemo(() => {
+    if (zoomLevel < 11) return { type: 'Feature' } as Feature
+    return {
+      type: 'Feature',
+      geometry: {
+        type: 'Polygon',
+        coordinates: bounds
+          ? [
+              [
+                [bounds[0][0], bounds[0][1]],
```

Comment:
this bounding box logic is a little obtuse, and I see it in a couple places. any way to extract it to a utility function?

Reply:
Extracted it to MapUtils